test/unit_tests/test_ahs_device.py

```diff
@@ -708,6 +770,207 @@ def test_run_task(self):
         assert len(task.result().measurements) == 17  # dev_sim takes 17 shots
         assert isinstance(task.result().measurements[0], ShotResult)
 
+    @staticmethod
+    def dummy_cfunc(t):
+        """Dummy function for testing local detunings"""
+        return 10
+
+    @pytest.mark.parametrize(
+        "pulses, expected_detunings",
+        [
+            ([HardwarePulse(0, 0, 2, [0, 1])], [2, 2, 0]),
+            (
+                [
+                    HardwarePulse(0, 0, dummy_cfunc, [0]),
+                    HardwarePulse(0, 0, dummy_cfunc, [1, 2]),
+                ],
+                [dummy_cfunc, dummy_cfunc, dummy_cfunc],
+            ),
+            (
+                [
+                    HardwarePulse(0, 0, 4, [0]),
+                    HardwarePulse(0, 0, 2, [1, 2]),
+                ],
+                [4, 2, 2],
+            ),
+            (
+                [HardwarePulse(0, 0, dummy_cfunc, [0, 1])],
+                [dummy_cfunc, dummy_cfunc, lambda t: 0],
+            ),
+        ],
+    )
+    def test_create_valid_local_detunings(self, pulses, expected_detunings):
+        """Test that BraketLocalAhsDevice._create_valid_local_detunings expands and pads the
+        detunings so that there is one local detuning per device wire, and that the detunings
+        are mapped correctly."""
+
+        detunings = _create_valid_local_detunings(pulses, dev_sim.wires)
+
+        assert len(detunings) == len(dev_sim.wires)
+
+        if callable(pulses[0].frequency):
+            for det, expected_det in zip(detunings, expected_detunings):
+                for i in range(10):
+                    assert det(i) == expected_det(i)
+        else:
+            assert all(detunings[i] == expected_detunings[i] for i in range(len(dev_sim.wires)))
```

Comment:
Instead of having this conditional logic in one test, can it be split into two test?